lighthouse-plugin-publisher-ads/audits/script-injected-tags.js

```diff
@@ -12,31 +12,33 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+const array = require('../utils/array.js');
 const i18n = require('lighthouse/lighthouse-core/lib/i18n/i18n');
+const NetworkRecords = require('lighthouse/lighthouse-core/computed/network-records');
 const PageDependencyGraph = require('lighthouse/lighthouse-core/computed/page-dependency-graph');
 const {auditNotApplicable} = require('../messages/common-strings');
 const {Audit} = require('lighthouse');
-const {getTimingsByRecord} = require('../utils/network-timing');
-const {getTransitiveClosure} = require('../utils/graph');
-const {isGptAdRequest} = require('../utils/resource-classification');
+const {getAdCriticalGraph} = require('../utils/graph');
+const {getScriptEvaluationTimes} = require('../utils/network-timing');
 
 /** @typedef {LH.Artifacts.NetworkRequest} NetworkRequest */
 /** @typedef {LH.Gatherer.Simulation.NodeTiming} NodeTiming */
 
+// Don't bother failing on scripts that are dynamically inserted but load
+// quickly.
+const MINIMUM_LOAD_TIME_MS = 400;
+
 const UIStrings = {
-  title: 'No script-injected tags found',
-  failureTitle: 'Avoid script-injected tags',
-  description: 'Load scripts directly with `<script async src=...>` instead ' +
-  'of injecting scripts with JavaScript. This allows the browser fetch the ' +
-  'script sooner. [Learn more](' +
+  title: 'Ad scripts are loaded statically',
+  failureTitle: 'Load ad scripts statically',
+  description: 'Load the following scripts directly with ' +
+  '`<script async src=...>` instead of injecting scripts with JavaScript. ' +
+  'Doing so allows the browser to preload scripts sooner. [Learn more](' +
   'https://www.igvita.com/2014/05/20/script-injected-async-scripts-considered-harmful/' +
   ').',
-  displayValue: '{numTags, plural, =1 {1 script-injected resource} ' +
-  'other {# script-injected resources}}',
-  columnRequest: 'Resource',
-  columnStartTime: 'Start',
-  columnDuration: 'Duration',
-  columnLineNumber: 'Source Line Number',
+  failureDisplayValue: 'Load {tags, plural, =1 {1 script} other {# scripts}} statically',
+  columnUrl: 'Script',
+  columnLoadTime: 'Load Time',
```

Comment:
Add to .json file for L10N? Or is that done automatically now?

Reply:
Automatically done now